### DISCUSSION TITLE
ring-mqtt on QNAP (or other NAS)

### DISCUSSION THREAD

**Discussion by Harry4242** [tsightler/ring-mqtt#631] [Support Q&A] [answered]
Hi,

i want to run ring-mqtt on my qnap NAS as a docker container.
the issue i am facing is the following.
i mapped the /data folder to my NAS --> worked
when i start the container i get the following problem..it starts up..say there is no ring-data.json..and the it shuts down the container before i can do anything.
i have only CLI access as long as the container runs..wich is only about 30sec....
is there a way to prevent the container shut down so i can start creating a token ? 

below is part of the log thats beeing created...

Thanks in advance !

Harry

ring-mqtt Detected runmode: docker +0ms
  ring-mqtt Configuration file: /data/config.json +19ms
  ring-mqtt State file /data/ring-state.json not found. No saved state data available. +0ms
  ring-mqtt No refresh token was found in the state file, use the Web UI at http://<host_ip_address>:55123/ to generate a token. +0ms
  ring-mqtt Succesfully started the token generator web UI +0ms
  ring-mqtt Unexpected token < in JSON at position 2 +16s
  ring-mqtt Configuration file could not be read, check that it exist and is valid. +3ms
[07:13:16] WARNING: A critical error was detected, shutting down container..

**Comment by Harry4242** [comment, marked as answer]
Never mind. i fixed it by creating  'fake' config.json..then the container stops and i can create a token.
Thanks !1

**Reply by tsightler** [reply]
I'm sure QNAP provides a way to run docker command interactively, in which case you just need to run the init-ring-mqtt.js as described in the documentation.
